### PR TITLE
Fix resource create and set commands to take in the passed in properties

### PR DESCRIPTION
**Pull request by vivsriaus**
The content to be added to Changelog is as follows:
* Fix resource create and set commands to take in the passed in properties

### CODE REVIEW

**Comment by amarzavery** [comment]
@vivsriaus - Can you please add tests for these? Or improve assertions in the current tests?

File: lib/commands/arm/resource/resource._js [hunk 1]
```diff
@@ -74,7 +74,7 @@ exports.init = function (cli) {
     .option('--subscription <subscription>', $('the subscription identifier'))
     .execute(function (resourceGroup, name, resourceType, properties, apiVersion, options, _) {
```
Comment (amarzavery) [inline lib/commands/arm/resource/resource._js]: @vivsriaus 
.execute(function (resourceGroup, name, resourceType, **properties**, apiVersion, options, _) {

- properties is defined explicitly over here. Why are you taking this from options (options.properties) ?

Please add a test or tighten the current tests. I am not sure that this is still working.

**Comment by vivsriaus** [comment]
@amarzavery Yes, tests are still pending. I was waiting for the node SDK PR to be merged, so I can add new tests with the new sdk

File: lib/commands/arm/resource/resource._js (hunk 1 above)
Comment (vivsriaus) [inline lib/commands/arm/resource/resource._js]: Added a test to ensure passed in properties is indeed honored

**Comment by amarzavery** [comment]
@vivsriaus - resolve conflicts

File: test/commands/arm/resource/arm.resource-tests.js [hunk 2]
```diff
@@ -65,13 +65,38 @@ describe('arm', function () {
...
+      });
+
       it('should work without switches', function (done) {
         var groupName = suite.generateId('xTestResource', createdGroups, suite.isMocked);
         var resourceName = suite.generateId('xTestGrpRes', createdResources, suite.isMocked);
         suite.execute('group create -n %s --location %s --json', groupName, testGroupLocation, function (result) {
           result.exitStatus.should.equal(0);
 
-          suite.execute('resource create %s %s %s %s %s --json', groupName, resourceName, 'Microsoft.Web/sites', testApiVersion, '{ "name": "' + resourceName + '", "siteMode": "Limited", "computeMode": "Shared" }', function (result) {
+          suite.execute('resource create %s %s %s %s -p %s --json', groupName, resourceName, 'Microsoft.Web/sites', testApiVersion, '{ "name": "' + resourceName + '", "siteMode": "Limited", "computeMode": "Shared" }', function (result) {
```
Comment (TianoMS) [inline test/commands/arm/resource/arm.resource-tests.js]: >-p %s [](start = 53, length = 5)

This test case seems to be testing "without switches". Do we want to add -p here?

File: lib/commands/arm/resource/resource._js [hunk 3]
```diff
@@ -52,7 +52,7 @@ exports.init = function (cli) {
     .option('--subscription <subscription>', $('the subscription identifier'))
     .execute(function (resourceGroup, name, resourceType, apiVersion, options, _) {
       var values = getValuesForCommand(resourceGroup, name, resourceType, apiVersion, options);
-      resource.createResource(values.resourceGroup, values.resourceName, values.resourceType, values.apiVersion, values.properties, values.options, _);
+      resource.createResource(values.resourceGroup, values.resourceName, values.resourceType, values.apiVersion, options.properties, values.options, _);
```
Comment (TianoMS) [inline lib/commands/arm/resource/resource._js]: >options.properties [](start = 113, length = 18)

Can we update "getValuesForCommand" so we can use values.properties?

**Pull request merged**